server/user/bitrot-detect

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+NCPUS=${1}
+if [[ -z "${NCPUS}" ]]; then
+    echo "Missing number of CPUs to use" >&2
+    exit 1
+fi
+
+TARGETDIR=${2}
+if [[ -z "${TARGETDIR}" ]]; then
+    echo "Missing target directory argument" >&2
+    exit 1
+fi
+
+WORKDIR=/var/tmp/bitrot-detect
+mkdir -p ${WORKDIR}
+
+# Clean up previous run.
+rm -f ${WORKDIR}/tbs.md5-*
+
+find ${TARGETDIR}/ -maxdepth 1 -type d -printf '%P\n' | sort | while read ctrl; do
+    cat ${TARGETDIR}/${ctrl}/*.md5 2> /dev/null | while read hash file; do
+        echo "${hash} ${TARGETDIR}/${ctrl}/${file}"
+    done
+done > ${WORKDIR}/tbs.md5 2> ${WORKDIR}/tbs.err
+
+cat ${WORKDIR}/tbs.err; echo ""
```

Comment:
You shouldn't need the empty argument to `echo` (it will produce the newline without it).

Ditto line 40.

Reply:
(And, if you're interested, I think there's a way to incorporate the `cat` into the previous command using `tee`, but it's a little bit woo-woo.... 😉)

Reply:
Agreed, though I left the empty string there to indicate it was intentional.

Reply:
I might actually be convinced to use `printf` in this particular case:
```
cat ${WORKDIR}/tbs.err | xargs -0 printf -- "%s\n"
```
😉